nexus/db-queries/src/db/datastore/disk.rs

```diff
@@ -657,7 +657,10 @@ impl DataStore {
     /// If the disk delete saga unwinds, then the disk should _not_ remain
     /// deleted: disk delete saga should be triggered again in order to fully
     /// complete, and the only way to do that is to un-delete the disk. Set it
-    /// to faulted to ensure that it won't be used.
+    /// to faulted to ensure that it won't be used. Use the disk's UUID as part
+    /// of its new name to ensure that even if a user created another disk that
+    /// shadows this "phantom" disk the original can still be un-deleted and
+    /// faulted.
```

Comment:
It might be worth calling out - no matter how silly - that like, technically the user could be using this name for another too. Seems unlikely though!

Reply:
You're not wrong! added something in 1ac073e8a